test/unit/change_trust_asset_test.js

```diff
@@ -0,0 +1,355 @@
+        Buffer.from([0, 0, 0, 0]).toString()
+      );
+    });
+
+    it('parses a 3-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'USD',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('USD\0');
+    });
+
+    it('parses a 4-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'BART',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('BART');
+    });
+
+    it('parses a 5-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        '12345',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum12');
+      expect(xdr.value().assetCode()).to.equal('12345\0\0\0\0\0\0\0');
+    });
+
+    it('parses a 12-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        '123456789012',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum12');
+      expect(xdr.value().assetCode()).to.equal('123456789012');
+    });
+
+    it('parses a liquidity pool trustline asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.liquidityPoolAsset({
+        assetA,
+        assetB,
+        fee
+      });
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(xdr.arm()).to.equal('liquidityPool');
+
+      const gotPoolParams = asset.getLiquidityPoolParameters();
+      expect(gotPoolParams.assetA).to.be.equal(assetA);
+      expect(gotPoolParams.assetB).to.be.equal(assetB);
+      expect(gotPoolParams.fee).to.be.equal(fee);
+    });
+  });
+
+  describe('fromOperation()', function() {
+    it('parses a native asset XDR', function() {
+      const xdr = new StellarBase.xdr.ChangeTrustAsset.assetTypeNative();
+      const asset = StellarBase.ChangeTrustAsset.fromOperation(xdr);
```

Comment:
This isn't the standard way to do this in the SDK, as far as I know. It should be doable via `Operation.fromXDRObject(xdr)`; do we support that?

Reply:
Oh, I see `Asset.fromOperation` now, so this is akin to that, gotcha